config/boards/xiaomi-umi.wip

```diff
@@ -0,0 +1,106 @@
+# Generate kernel and rootfs image for Qcom ABL booting
+declare -g BOARD_NAME="Xiaomi Umi"
+declare -g BOARD_MAINTAINER="FantasyGmm"
+declare -g BOARDFAMILY="qcom-abl"
+declare -g KERNEL_TARGET="sm8250"
+declare -g KERNELPATCHDIR="arm64-sm8250"
+declare -g EXTRAWIFI="no"
+declare -g MODULES="spi-geni-qcom"
+declare -g BOOTCONFIG="none"
+declare -g IMAGE_PARTITION_TABLE="gpt"
+declare -g -a ABL_DTB_LIST=("sm8250-xiaomi-umi")
+
+# Use the full firmware, complete linux-firmware plus Armbian's (for qcom/a650_sqe.fw)
+declare -g BOARD_FIRMWARE_INSTALL="-full"
+
+function post_family_config_branch_sm8250__pmos_kernel() {
+	display_alert "Setting up kernel for" "${BOARD}" "info"
+	declare -g KERNEL_MAJOR_MINOR="6.7" # Major and minor versions of this kernel.
+	declare -g KERNELBRANCH='branch:linux-6.7.y'
+}
+
+function xiaomi-umi_is_userspace_supported() {
+	[[ "${RELEASE}" == "jammy" ]] && return 0
+	[[ "${RELEASE}" == "trixie" ]] && return 0
+	[[ "${RELEASE}" == "noble" ]] && return 0
+	return 1
+}
+
+function post_family_tweaks_bsp__xiaomi-umi_firmware() {
+	display_alert "$BOARD" "Install firmwares for xiaomi umi" "info"
+
+	# Alsa-ucm-conf profile for Xiaomi Umi
+	mkdir -p $destination/usr/share/alsa/ucm2/conf.d/sm8250
+	install -Dm644 $SRC/packages/bsp/xiaomi-umi/umi.conf $destination/usr/share/alsa/ucm2/Xiaomi/umi/umi.conf
+	install -Dm644 $SRC/packages/bsp/xiaomi-umi/umi_HiFi.conf $destination/usr/share/alsa/ucm2/Xiaomi/umi/HiFi.conf
+	ln -sfv ../../Xiaomi/umi/umi.conf \
+		"$destination/usr/share/alsa/ucm2/conf.d/sm8250/Xiaomi Mi 10.conf"
+
+	# USB Gadget Network service
+	mkdir -p $destination/usr/local/bin/
+	mkdir -p $destination/usr/lib/systemd/system/
+	install -Dm655 $SRC/packages/bsp/xiaomi-umi/setup-usbgadget-network.sh $destination/usr/local/bin/
+	install -Dm655 $SRC/packages/bsp/xiaomi-umi/remove-usbgadget-network.sh $destination/usr/local/bin/
+	install -Dm644 $SRC/packages/bsp/xiaomi-umi/usbgadget-rndis.service $destination/usr/lib/systemd/system/
```

Comment:
These usbgadget network scripts are the same as thoese from xiaomi-elish, what about moving them to a common one? Then we don't have to write almost the same scripts for each device.